src/dynamic_batch_scheduler.cc

```diff
@@ -438,9 +443,16 @@ DynamicBatchScheduler::GetDynamicBatch()
   // batch size would be exceeded or if the shape of the next request
   // does not match the shape of the pending batch.
   bool send_now = false;
+
+  // If the previous payload was not executed, reset the cursor to the start
+  // of the queue to re-iterate over it and find the ideal batch.
   if (!queue_.IsCursorValid()) {
     queue_.ResetCursor();
     pending_batch_size_ = 0;
+    if (CustomBatchEnabled()) {
+      CustomBatchFini();
+      CustomBatchInit();
+    }
```

Comment:
Can you elaborate on this? Based on the comment you added, doesn't that mean the `payload` is scheduled but we still keep it as `curr_payload_` and attempt to add requests onto the payload? Then shouldn't we keep the current custom batch instead of starting a new one?

Reply:
If the payload was not executed, curr_payload_ starts from scratch. It resets the cursor, so it starts building the payload from zero requests (does not keep current requests).